spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfiguration.java

```diff
@@ -85,4 +92,18 @@ public TaskProperties taskProperties() {
 	DataflowTaskExecutionDao dataflowTaskExecutionDao(DataSource dataSource, TaskProperties taskProperties) {
 		return new JdbcDataflowTaskExecutionDao(dataSource, taskProperties);
 	}
+
+	@Bean
+	DataflowTaskExecutionMetadataDao dataflowTaskExecutionMetadataDao(DataSource dataSource, ApplicationContext context) {
+		DataFieldMaxValueIncrementerFactory incrementerFactory = new DefaultDataFieldMaxValueIncrementerFactory(dataSource);
+		String databaseType;
+		try {
+			databaseType = DatabaseType.fromMetaData(dataSource).name();
```

Comment:
Looks like batch doesn't understand a real mariadb database so this wont work if mariadb is treated as its own db type and not just mysql compatible.

Reply:
We don't treat Maria that way now (and this is the task version of the class, not batch) so is it an issue?

Reply:
I'd like to be able to remove a requirement for that mariadb driver workaround that it fakes db to be mysql for real mariadb. Essentially spring-cloud-task needs to do same dance as rest of a world by starting to think those two as a separate db types. Also newer mariadb's now has sequences, etc. This comment was more of a reminder for now.

Reply:
Ok...I'd definitely consider that dance out of scope for this PR.